packages/forms-web-app/src/controllers/full-appeal/submit-appeal/letter-confirming-application.js

```diff
@@ -68,8 +69,15 @@ const postLetterConfirmingApplication = async (req, res) => {
 			};
 		}
 
-		appeal.sectionStates[sectionName][taskName] = COMPLETED;
-		req.session.appeal = await createOrUpdateAppeal(appeal);
+		if (req.body['save-and-return'] !== '') {
+			appeal.sectionStates[sectionName][taskName] = COMPLETED;
+			req.session.appeal = await createOrUpdateAppeal(appeal);
+
+			return res.redirect(`/${TASK_LIST}`);
+		} else {
+			appeal.sectionStates[sectionName][taskName] = IN_PROGRESS;
+			return await postSaveAndReturn(req, res);
```

Comment:
do you need `req.session.appeal = await createOrUpdateAppeal(appeal)` here?

Reply:
ah great catch, cheers @edpackard-work, updated accordingly!